activesupport/lib/active_support/error_reporter.rb

```diff
@@ -166,6 +166,8 @@ def set_context(...)
     #   Rails.error.report(error)
     #
     def report(error, handled: true, severity: handled ? :warning : :error, context: {}, source: DEFAULT_SOURCE)
+      return if error.instance_variable_get(:@__rails_error_reported)
```

Comment:
Just off the top of my head: do you think it'll help if Rails logs `#report` invocations in `log` level messages? It can:

1. Hint users about duplicated reporting in their apps, whether intentional or by accident (usually the latter)
2. Let users know there are exceptions being reported under the code path, which they sometimes don't know until production because of how SDKs work. 
    - Feels like it could help both debug error reporting issues and also general debugging? 

What I've been thinking is, the invocation of error reporter could be another way to observe and debug Rails applications. And before this change, both of usages can be achieved by subscribing a `DebugSubscriber` (whether provided by the framework or others). 

But with this change, that approach won't detect 1 anymore. So if we want to log all the invocation of `#report`, we need to do it here. (Which also makes sense since Rails already provides many useful logging messages).

Reply:
Yeah, I thought about this a bit previously. What I had in mind was to have a default subscriber in dev and test environments that simply log.

That said your point about double logging is interesting, but I think it can be handled ad hoc, e.g. `Rails.logger.warning("error already reported .....")`

Reply:
Merging as is for now, we can do the logging in a follow-up.